javascript/ReportAdmin.js

```diff
@@ -2,17 +2,32 @@
  * File: ReportAdmin.js
  */
 
-(function($) {
-	$.entwine('ss', function($){
-		$('.ReportAdmin .cms-edit-form').entwine({
-			onsubmit: function(e) {
-				var url = $.path.parseUrl(document.location.href).hrefNoSearch,
-					params = this.find(':input[name^=filters]').serializeArray();
-				params = $.grep(params, function(param) {return (param.value);}); // filter out empty
-				if(params) url = $.path.addSearchParams(url, $.param(params));
-				$('.cms-container').loadPanel(url);
-				return false;
-			}
-		});
-	});
+(function ($) {
+  $.entwine("ss", function ($) {
+    $(".ReportAdmin .cms-edit-form").entwine({
+      onsubmit: function (e) {
+        let url = $.path.parseUrl(document.location.href).hrefNoSearch;
+        let params = this.find(":input[name^=filters]").serializeArray();
+
+        try {
+          params = $.grep(params, function (param) {
+            // filter out empty
+            return param.value;
+          });
+
+          // convert params to a query string
+          params = $.param(params);
+
+          // append query string to url
+          url += "?" + params;
```

Comment:
I know we have a URL here with no query string, but concatenating like this just doesn't quite feel right... how would you feel about using [the `URL` API](https://developer.mozilla.org/en-US/docs/Web/API/URL) instead?

```js
// replace let url = $.path.parseUrl(document.location.href).hrefNoSearch; with this:
let url = new URL(document.location.href);
url.search = '';

// replace url += "?" + params; with this:
url.search = params;

// and then in loadPanel pass in the url like this:
$('.cms-container').loadPanel(url.toString());
```

Reply:
Hi @GuySartorelli, that fails. Generates admin/reports/show/TestReport?[object%20Object]

Reply:
It shouldn't do - params is a string by the time `url.search = params;` is called... did you put the snippets where the comments say to put them? I could see how it would be easy to just put what I've got there directly instead, which would indeed have that incorrect result.

Reply:
Yep, definitely doesn't work. Also tested via URLSearchParams and that also didn't handle multple fields.

Reply:
Hmm. I'll have to take another look 9n Monday then cause that did work for me... but I'll probably end up merging this as is cause it works.

Reply:
Didn't end up looking at why my suggestion didn't work - instead I looked at the PR again and decided I'm happy with it.